File: 1780-check-if-number-is-a-sum-of-powers-of-three/1780. Check if Number is a Sum of Powers of Three.java
class Solution {
    public boolean checkPowersOfThree(int n) {
        
        /* Number to binary, instead of using 2, we are using 3 as the base here */
        int num = n;
        while (num > 0) {
            if (num % 3 > 1) {
                return false;
            }
            num = num / 3;
        }
        return true;
    }
}

File: 1780-check-if-number-is-a-sum-of-powers-of-three/1780. Check if Number is a Sum of Powers of Three.cpp
class Solution {
public:
    bool checkPowersOfThree(int n) {
        int k;
        for(int i=0;i<n;i++)
        {
            if(pow(3,i)>n)
            {
                k = i-1;
                break;

            }

            else if(pow(3,i)==n)
            {
                return true;
            }
        }

        cout<<k<<" "<<endl;
     while(n && k>=0)
     {
        if(n>=pow(3,k))
        {
            n=n-pow(3,k);
            cout<<n<<" n "<<endl;
        }
        k--;
     }

       if(n==0) return true;
       return false;
        
    }
};